finality-aleph/src/base_protocol/service.rs

```diff
@@ -0,0 +1,115 @@
+use std::{
+    fmt::{Display, Formatter, Result as FmtResult},
+    sync::{
+        atomic::{AtomicBool, AtomicUsize},
+        Arc,
+    },
+};
+
+use futures::stream::StreamExt;
+use log::{debug, trace, warn};
+use sc_network_common::sync::SyncEvent;
+use sc_network_sync::{service::chain_sync::ToServiceCommand, SyncingService};
+use sc_utils::mpsc::{tracing_unbounded, TracingUnboundedReceiver, TracingUnboundedSender};
+use sp_runtime::traits::{Block, Header};
+
+use crate::{
+    base_protocol::{handler::Handler, LOG_TARGET},
+    BlockHash, BlockNumber,
+};
+
+#[derive(Debug)]
+pub enum Error {
+    NoIncomingCommands,
+}
+
+impl Display for Error {
+    fn fmt(&self, f: &mut Formatter<'_>) -> FmtResult {
+        use Error::*;
+        match self {
+            NoIncomingCommands => write!(f, "Channel with commands from user closed."),
+        }
+    }
+}
+
+/// A service that needs to be run to have the base protocol of the network work.
+pub struct Service<B>
+where
+    B: Block<Hash = BlockHash>,
+    B::Header: Header<Number = BlockNumber>,
+{
+    handler: Handler,
+    commands_from_user: TracingUnboundedReceiver<ToServiceCommand<B>>,
+    events_for_users: Vec<TracingUnboundedSender<SyncEvent>>,
+}
+
+impl<B> Service<B>
+where
+    B: Block<Hash = BlockHash>,
+    B::Header: Header<Number = BlockNumber>,
+{
+    /// Create a new service.
+    // TODO(A0-3886): This shouldn't need to return the substrate type after replacing RPCs.
+    // In particular, it shouldn't depend on `B`. This is also the only reason why
+    // the `major_sync` argument is needed.
+    pub fn new(major_sync: Arc<AtomicBool>) -> (Self, SyncingService<B>) {
```

Comment:
Syncing Engine also returned the `NonDefaultConfig` but I understand that we are going to create it somewhere else (in `bin/service` for instance)

Reply:
I'm not sure where we will end up creating it, but this is definitely part of the `NotificationEvent` task, since this is what interacts with it – that's what you need to get the notification events.